### PR TITLE
[DEAD-CODE] Added failing ShadowedLocalVariable test and implementation

### PR DESCRIPTION
**Pull request by clxswalther**
Variable assignment `$toDate = null;` should be removed because afterwards it always gets overwritten.

https://getrector.org/demo/21e1de43-d5cb-4f14-a0d7-ddaf9dbb5292

Closes https://github.com/rectorphp/rector/issues/5873

### CODE REVIEW

**Comment by staabm** [comment]
@clxswalther seems conflicts need to be resolved

**Comment by TomasVotruba** [comment]
Full feature is needed https://github.com/rectorphp/rector/issues/5873#issuecomment-816742961

**Comment by staabm** [comment]
thx for the feedback.

> Full feature is needed [#5873 (comment)](https://github.com/rectorphp/rector/issues/5873#issuecomment-816742961)

could you elaborate a bit? this PR contains a unit test and a fix.. what do you miss?

**Comment by TomasVotruba** [comment]
Ah, I read the title and assumed it's what is described in it :)
Also the CI was failing. 

I'll reopen, but we'll need CI green

**Comment by fbecker-complex** [comment]
@TomasVotruba
We added a fix for the provided failing test case.
Could you please review it.

p.s.
We fixed a phpstan error in `rules/Downgrade72/Rector/FuncCall/DowngradeStreamIsattyRector.php->createClosure($node)`, because there was no typehint provided for `$node`.

**Comment by TomasVotruba** [comment]
Thanks :+1: 

Could you rebase to remove conflicts? I'll review the PR then

File: rules/Downgrade72/Rector/FuncCall/DowngradeStreamIsattyRector.php [hunk 1]
```diff
@@ -135,7 +135,7 @@ private function createIf(Expr $expr): If_
         return $if;
     }
 
-    private function createClosure($node): Closure
+    private function createClosure(FuncCall $node): Closure
```
Comment (samsonasik) [inline rules/Downgrade72/Rector/FuncCall/DowngradeStreamIsattyRector.php:138]: You can update `$node` to `$funcCall` and the conflict will be fixed.

**Comment by TomasVotruba** [comment]
I've resolved conflicts here

**Pull request merged**